### PR TITLE
readme first draft

### PR DESCRIPTION
**Pull request by bdemann**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,9 +1,236 @@
...
+ToIdent is only for code simplification. We decided that we liked the look of my_string.to_ident() better than format_ident!("{}", my_string). So that's what we are doing.
+
+### Security
+
+Things to keep in mind:
+
+- The CDK Framework does not yet have extensive automated property tests
+- The CDK Framework does not yet have multiple independent security reviews/audits
+
+### Contributing
```
Comment (bdemann) [inline README.md:234]: Get rid of contributing section

File: README.md [hunk 2]
```diff
@@ -1,9 +1,236 @@
...
+DFX_VERSION=0.12.1 sh -ci "$(curl -fsSL https://sdk.dfinity.org/install.sh)"
+```
+
+#### Your CDK
+
+Your CDK should produce a rust project that has at least the following dependencies
+
+```toml
+[dependencies]
+ic-cdk = {{ version = "0.6.8", features = ["timers"] }}
```
Comment (bdemann) [inline README.md:53]: this is python it needs to be toml

File: README.md [hunk 3]
```diff
@@ -1,9 +1,236 @@
...
+  - [ToTypeAnnotation](#to-type-annotation)
+  - [Traits](#traits)
+- [Security](#security)
+- [Contributing](#contributing)
+
+## Usage
+
+### Installation
+
+You should be using a \*nix environment (Linux, Mac OS, [WSL](https://learn.microsoft.com/en-us/windows/wsl/install)) with bash and have Rust and dfx installed on your system.
```
Comment (bdemann) [inline README.md:37]: get rid of all of this down to the the your cdk

File: README.md
Comment (bdemann) [inline README.md:1]: spell check pass

File: README.md [hunk 4]
```diff
@@ -1,9 +1,236 @@
...
+    // TODO write contents of lib_file to a file
+
+    // TODO generate cargo.toml
+
+    // TODO build your generated rust project and deploy to replica
+}
+
+```
+
+### Candid Types
```
Comment (bdemann) [inline README.md:150]: Match the order, and have a little bit better of a transition saying that we are now explaining each for the fields. Give a small table of contents

File: README.md [hunk 5]
```diff
@@ -1,9 +1,236 @@
...
+ic-cdk = {{ version = "0.6.8", features = ["timers"] }}
+ic-cdk-macros = "0.6.8"
+candid = "0.8.4"
+```
+
+Create a Rust project with the following dependencies \*Note: For rev use the hash of current commit on the main branch
+
+```toml
+[dependencies]
+cdk_framework = { git = "https://github.com/demergent-labs/cdk_framework", rev = "08a74a4077d0981acde2b0fc714e877bba659b9e" }
```
Comment (bdemann) [inline README.md:62]: Point out that its not on crates.io
Move this to the top

File: README.md [hunk 6]
```diff
@@ -1,9 +1,236 @@
...
+Your CDK should produce a rust project that has at least the following dependencies
+
+```toml
+[dependencies]
+ic-cdk = {{ version = "0.6.8", features = ["timers"] }}
+ic-cdk-macros = "0.6.8"
+candid = "0.8.4"
+```
+
+Create a Rust project with the following dependencies \*Note: For rev use the hash of current commit on the main branch
```
Comment (bdemann) [inline README.md:58]: Look for git hub note syntax. apply this to all notes

File: README.md [hunk 7]
```diff
@@ -1,9 +1,236 @@
...
+```toml
+[dependencies]
+cdk_framework = { git = "https://github.com/demergent-labs/cdk_framework", rev = "08a74a4077d0981acde2b0fc714e877bba659b9e" }
+quote = "1.0.21"
+proc-macro = "1.0.43"
+```
+
+The CDK framework will help to generate the lib.rs file for you canister, but you cdk will still need to provide everything else.
+
+### Abstract Canister Tree
```
Comment (bdemann) [inline README.md:69]: Move the core concept to here and elaborate more.

File: README.md [hunk 8]
```diff
@@ -1,9 +1,236 @@
...
+HasInlines which is a helper for collecting inline types. It is implemented on two of the other traits, IsCallable and HasMembers, which anything that is callable or has members can implement to make it easier to collect the inline types for that node. For example the external_canister::Method implements IsCallable. That implementation will allow IsCallable to know how to access it's return type and params, so that you can call flatten_inlines (from HasInlines) directly on the external_canister::Method to get all of the inline types for the Member.
+
+The HasInlineName trait is just to make a consistent interface for things that may have a special inline name. For example the ReturnType needs to append "ReturnType" to the end of all it's names. The HasInlineName implementation for ReturnType will take care of that so that it can be applied consistently everywhere.
+
+ToIdent is only for code simplification. We decided that we liked the look of my_string.to_ident() better than format_ident!("{}", my_string). So that's what we are doing.
+
+### Security
```
Comment (lastmjs) [inline README.md:227]: Remove this section

File: README.md [hunk 9]
```diff
@@ -1,9 +1,236 @@
...
+
+    // TODO write contents of lib_file to a file
+
+    // TODO generate cargo.toml
+
+    // TODO build your generated rust project and deploy to replica
+}
+
+```
+
```
Comment (lastmjs) [inline README.md:149]: Explain here that we're going to dive into each field of the ACT

Comment (lastmjs) [reply README.md:149]: Create a glossary/index thing for each field of the ACT

File: README.md [hunk 10]
```diff
@@ -1,9 +1,236 @@
...
+### Installation
+
+You should be using a \*nix environment (Linux, Mac OS, [WSL](https://learn.microsoft.com/en-us/windows/wsl/install)) with bash and have Rust and dfx installed on your system.
+
+#### dfx
+
+Run the following command to install dfx 0.12.1:
+
+```bash
+DFX_VERSION=0.12.1 sh -ci "$(curl -fsSL https://sdk.dfinity.org/install.sh)"
```
Comment (lastmjs) [inline README.md:44]: Probably get rid of this

File: README.md [hunk 11]
```diff
@@ -1,9 +1,236 @@
...
-The CDK Framework helps CDK authors build CDKs.
+The CDK Framework helps CDK authors build CDKs for the [Internet Computer](https://internetcomputer.org/).
 
 ## Core Concepts
 
 Start with the AST of your language and compile it into an Abstract Canister Tree (a concept from the CDK Framework). The CDK Framework will then create a Rust canister which can be compiled into Wasm and deployed to the IC.
 
 AST -> ACT -> Rust canister code -> Wasm binary -> IC
+
+## Disclaimer
```
Comment (lastmjs) [inline README.md:11]: Remove this

File: README.md [hunk 12]
```diff
@@ -1,9 +1,236 @@
 # CDK Framework
 
-The CDK Framework helps CDK authors build CDKs.
+The CDK Framework helps CDK authors build CDKs for the [Internet Computer](https://internetcomputer.org/).
 
 ## Core Concepts
 
 Start with the AST of your language and compile it into an Abstract Canister Tree (a concept from the CDK Framework). The CDK Framework will then create a Rust canister which can be compiled into Wasm and deployed to the IC.
```
Comment (lastmjs) [inline README.md:7]: Flesh this out a bit to explain that the ACT will generate a `lib.rs` file, and explain that a Cargo.toml will need to be generated...or somehow explain that the lib.rs file can then be used to create a Rust canister

**Review by lastmjs — CHANGES_REQUESTED** [review]

File: README.md [hunk 13]
```diff
@@ -1,9 +1,236 @@
...
+
+#### dfx
+
+Run the following command to install dfx 0.12.1:
+
+```bash
+DFX_VERSION=0.12.1 sh -ci "$(curl -fsSL https://sdk.dfinity.org/install.sh)"
+```
+
+#### Your CDK
```
Comment (dansteren) [inline README.md:47]: This section is good, but doesn't belong in the installation section. It should be in a separate place. Like, they need to make an ACT, and generate a Cargo.toml, etc. That's where this belongs.

**Review by dansteren — CHANGES_REQUESTED** [review]

**Pull request merged**